### ISSUE TITLE
How to add new layouts to the subnavigation.

### ISSUE THREAD

**Issue by JavierVelazquez** [benjaminkott/bootstrap_package#867]
# Support

## Description
Hello all!! Maybe someone can help me: I have made a little extension to the bootstrap package to add a couple of layouts and maybe, if I manage, a couple of new content elements. So far I can make the layouts appear in the back end and the front end. Some only on the back... (it beats me why they don't work even when I make them identical) but my question is how do you make this layouts appear in the subnavigation (the one one the side column) I am slowly getting better but I find it really hard. Typo3 is very dense sometimes. Anyway, I hope everyone is staying healthy and cheerful. Thank you!!
[Is your question related to a problem? Please describe.]

Basically if the page doesn't use a Bootstrap Package layout, it doesn't get included. I guess is because my extension goes after the BS Package. So, how does one go about connecting them?
ANy help greatly appreciated

**Comment by gilbertsoft** [comment]
Do you know https://www.sitepackagebuilder.com? The generated sitepackage extension includes a sample BE layout. Make sure you clear all caches after a change, that should do it. I'm not sure if you talk about a sitepackage extension or a common one to use in multiple projects:

- For the first case have a look at the sitepackagebuilder, there `bootstrap_package` gets included in the TypoScript constants and setup and you need to adjust your template, include your sitepackage instead of `bootstrap_package`.
- If you are talking about a common extension it finally depends if you like to have a dependency to `bootstrap_package` or not. If not you should include your extension after `bootstrap_package`.

**Comment by JavierVelazquez** [comment]
Hello Gilbersoft, Thank you very much for the answer! I did start building the extension from https://www.sitepackagebuilder.com and the layout on the extension plus another one I added do work: they appear on the back end, and they render in the front end. 
Another two more layouts do appear in the back end but do not render (they throw an error code when opening the pages) so that is one issue and a weir one because the code is the same...
Now, if I understand well, the dependency is there (I just double checked) right from the start when I built the extension on the web because I choose bootstrap_package over fluid_style_content.
My problem (apart from the layout) is that somehow the submenu doesn't "read" pages that I add that use the new layouts. The top menu picks them just fine but the subnavigation (the side one in certain layouts) doesn't. 
Am I missing some other link that has to be done? apart from the dependency? When I place the bootstrap-package after the extension in the template it all breaks apart, by the way, so I have my mini extension after bs package. 
I do clear all caches all the time, my kingdom for a button that saves, closes and clears the cache at once... ( I do have the save & close one...)
Thanks a lot for even reading this!!
